src/util/ogr_source_util.cpp

```diff
@@ -533,13 +619,55 @@ bool OGRSourceUtil::readTimeIntoCollection(const QueryRectangle &rect, OGRFeatur
 	}
 	
 	//check if features time stamps have overlap with query rectangle times
-	if(t1 < rect.t1 && t2 < rect.t1 || t1 > rect.t1 && t2 > rect.t2)
+	if(t1 < rect.t1 && t2 < rect.t1 || t1 > rect.t2 && t2 > rect.t2)
 		return false;
 
 	time.emplace_back(TimeInterval(t1, t2));
 	return true;
 }
 
+time_t OGRSourceUtil::getFieldAsTime_t(OGRFeature *feature, int featureIndex){
+    using namespace boost::posix_time;
+
+    int d = 0, mon = 0, y = 0, h = 0, min = 0, s = 0, time_zone = 0;
+    bool success = feature->GetFieldAsDateTime(featureIndex, &y, &mon, &d, &h, &min, &s, &time_zone);
+    if(!success)
+        throw OperatorException("OGRSource: Could not read time attribute.");
+
+    ptime time(boost::gregorian::date(y, mon, d), hours(h)+minutes(min)+seconds(s));
+
+    // Return values for time_zone flag are explained here:
+    // http://www.gdal.org/classOGRFeature.html#a9cbf2fc45b5674265db25183aa93b36c
+    if(time_zone == 0) //unknown, assume it as UTC
+    {
+        return to_time_t(time);
+    }
+    else if(time_zone == 1) //localtime
+    {
+		// Because we don't know when this happens and what it is supposed to mean in this context
+        // (local computer time, or the local time where the data was recorded [what we don't know]) we throw an exception for now.
+		throw OperatorException("OGRSource: Unexpected time zone value from the time attribute: \"localtime\"");
+    }
+    else
+    {
+        // time zone is explicitly given, 100 <-> +00, 104 <-> +01; 108 <-> +02; 96 <-> -01, etc
+        int zone_hours = (time_zone - 100) / 4;
```

Comment:
what about fractional time zones? https://www.timeanddate.com/time/time-zones-interesting.html

Reply:
Support is added.